lib/Serialization/Deserialization.cpp

```diff
@@ -3418,15 +3428,27 @@ class swift::DeclDeserializer {
     GenericEnvironmentID genericEnvID;
     TypeID superclassID;
     uint8_t rawAccessLevel;
-    unsigned numConformances;
-    ArrayRef<uint64_t> rawInheritedIDs;
+    unsigned numConformances, numInheritedTypes;
+    ArrayRef<uint64_t> rawInheritedAndDependencyIDs;
     decls_block::ClassLayout::readRecord(scratch, nameID, contextID,
                                          isImplicit, isObjC,
                                          requiresStoredPropertyInits,
                                          inheritsSuperclassInitializers,
                                          genericEnvID, superclassID,
                                          rawAccessLevel, numConformances,
-                                         rawInheritedIDs);
+                                         numInheritedTypes,
+                                         rawInheritedAndDependencyIDs);
+
+    Identifier name = MF.getIdentifier(nameID);
+
+    for (TypeID dependencyID :
```

Comment:
Is it right for inheritance clause entries to add dependencies? The change I'm working on just drops ones that don't deserialize. I think only the superclass and raw type of an enum should introduce dependencies.

Reply:
The inheritance clause doesn't add dependencies; it's just the silly LLVM bitstream container format that forces us to put both dependencies and inherited types in the same array. That's what the `slice` is for.

Reply:
ah, oops.